docs/spec_draft.md

```diff
@@ -1808,6 +1809,68 @@ where `i[d] = j[permutation[d]]`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.while
+
+### Semantics
+
+Produces the output from executing `body` function 0 or more times till the
+`cond` function outputs `true`. More formally, the semantics can be expressed
+using Python-like syntax as follows:
+
+```python
+internal_state = operands
+while cond(internal_state) == True:
+  internal_state = body(internal_state)
+results = internal_state
+```
+
+The behaviour of an infinite loop is undefined.
+
+### Inputs
+
+| Name       | Type                                             |
+|------------|--------------------------------------------------|
+| `operands` | variadic number of tensors of any supported type |
+| `cond`     | `function`                                       |
+| `body`     | `function`                                       |
+
+### Outputs
+
+| Name      | Type                                             |
+|-----------|--------------------------------------------------|
+| `results` | variadic number of tensors of any supported type |
+
+### Constraints
+
```

Comment:
#365

Reply:
`if` and `case` can have time delay loops within them (if time delay is a use case), specifically with while since 0 outputs mean 0 inputs to while and 0 inputs to condition and body as well, a legitimate use case that is not an empty or an infinite loop is hard to imagine.